backend/src/main/java/com/huellapositiva/domain/service/ContactPersonProfileService.java

```diff
@@ -0,0 +1,109 @@
+                                                          String accountId) {
+        JpaContactPerson jpaContactPerson = jpaContactPersonRepository.findByAccountIdWithProfile(accountId)
+                .orElseThrow(() -> new UserNotFoundException("Contact person not found. Account ID: " + accountId));
+        boolean isNewEmail = !jpaContactPerson.getCredential().getEmail().equalsIgnoreCase(profileRequestDto.getEmail());
+        validations(profileRequestDto, isNewEmail);
+
+        if (isNewEmail) {
+            Role newJpaRole = jpaRoleRepository.findByName(CONTACT_PERSON_NOT_CONFIRMED.toString())
+                    .orElseThrow(() -> new RoleNotFoundException("Role " + CONTACT_PERSON_NOT_CONFIRMED.toString() + "not found."));
+            Set<Role> newUserRoles = new HashSet<>();
+            newUserRoles.add(newJpaRole);
+            jpaContactPerson.getCredential().setRoles(newUserRoles);
+            jwtService.revokeAccessTokens(jpaContactPerson.getCredential().getId());
+        }
+
+        updateContactPersonProfile(profileRequestDto, jpaContactPerson);
+        jpaContactPerson.getCredential().setEmail(profileRequestDto.getEmail());
+
+        jpaContactPersonRepository.save(jpaContactPerson);
+
+        return new UpdateProfileResult(isNewEmail);
+    }
+
+    /**
+     * Validate profile data.
+     *
+     * @param profileRequestDto New contact person profile information to update
+     * @param isNewEmail True if the contact person is updating the email
+     */
+    public void validations(UpdateContactPersonProfileRequestDto profileRequestDto, boolean isNewEmail){
+        if (isNewEmail && jpaCredentialRepository.findByEmail(profileRequestDto.getEmail()).isPresent()) {
+            throw new EmailAlreadyExistsException("Email already exists in the database.");
+        }
+        if (isNotPhoneNumber(profileRequestDto.getPhoneNumber())) {
+            throw new InvalidFieldException("The phone number field is invalid");
+        }
+    }
+
+    /**
+     *  Update information in profile table
+     *
+     *  @param profileRequestDto New contact person credential information to update
+     *  @param jpaContactPerson  JPA representation of the contact person
+     */
+    private void updateContactPersonProfile(UpdateContactPersonProfileRequestDto profileRequestDto,
+                                            JpaContactPerson jpaContactPerson) {
+        String id = jpaContactPerson.getContactPersonProfile().getId();
+        Integer surrogateKey = jpaContactPerson.getContactPersonProfile().getSurrogateKey();
+
+        JpaContactPersonProfile jpaProfile = JpaContactPersonProfile.builder()
+                .surrogateKey(surrogateKey)
+                .id(id)
+                .name(profileRequestDto.getName())
+                .surname(profileRequestDto.getSurname())
+                .phoneNumber(profileRequestDto.getPhoneNumber())
+                .build();
+        jpaContactPerson.setContactPersonProfile(jpaProfile);
```

Comment:
We should not create a new JpaXXX object every time we update it. We have to check first if that object exists and load it from the database in a JPA context otherwise we can get unexpected data inconsistency.

```
JpaContactPersonProfile profile = jpaContactPerson.getContactPersonProfile();
if (profile == null) {
  jpaProfile = JpaContactPersonProfile.builder()
                .id(Id.newId().getValue())
                .build();
  jpaContactPerson.setContactPersonProfile(jpaProfile);
}
jpaProfile.setName(...)
jpaProfile.setSurname(...)
jpaProfile.setPhoneNumber(...)
```

You could even skip the null check logic as we have declared the relation mandatory in `JpaContactPerson`
```
@OneToOne(optional = false)
private JpaContactPersonProfile contactPersonProfile;
```
Every time we create a JpaContactPerson we persist a JpaContactPersonProfile as well so there is no chance to have a ContactPerson in the database without a profile.

Reply:
Done